### PR TITLE
Backend wip

### PR DESCRIPTION
**Pull request by pmacaluso3**
The `models:workflow` rake task tells the whole story. This covers the workflow up through when the students visit each others' sites. Next step for me is to support messaging.

### CODE REVIEW

File: app/lib/build_science_fair.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  def initialize(cohort, params)
+    params = params.with_indifferent_access
+
+    @cohort = cohort
+    @students = cohort.students
+    @date = params[:date]
+    @name = params[:name]
+    @type = params[:type]
+    @num_rounds = params[:num_rounds]
```
Comment (jlr7245) [inline app/lib/build_science_fair.rb]: can prob get rid of this

File: app/models/user.rb [hunk 2]
```diff
@@ -1,8 +1,24 @@
 class User < ApplicationRecord
-  has_secure_password
-  enum user_type: %w(student instructor)
+  module UserTypes
+    STUDENT = 'student'
+    INSTRUCTOR = 'instructor'
```
Comment (jlr7245) [inline app/models/user.rb:4]: nice nice

File: db/migrate/20200923203802_create_tour_visits.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateTourVisits < ActiveRecord::Migration[6.0]
+  def change
+    create_table :tour_visits do |t|
+      t.decimal :duration
```
Comment (jlr7245) [inline db/migrate/20200923203802_create_tour_visits.rb:4]: did we wanna change this to an `end_time` timestamp? feels kinda dumb to have an `end_time` and an `updated_at` but ¯\_(ツ)_/¯

**Review by jlr7245 — APPROVED** [review]
lgtm, couple small pieces to change

File: db/migrate/20200923203802_create_tour_visits.rb (hunk 3 above)
Comment (pmacaluso3) [reply db/migrate/20200923203802_create_tour_visits.rb:4]: it's overwritten in a subsequent migration!

File: app/lib/build_science_fair.rb (hunk 1 above)
Comment (pmacaluso3) [reply app/lib/build_science_fair.rb]: trueee

**Pull request merged**